onelogin_rules/onelogin_high_risk_login.py

```diff
@@ -0,0 +1,43 @@
+import time
+from panther_oss_helpers import get_counter, increment_counter, set_key_expiration, reset_counter  # pylint: disable=import-error
+
+THRESH = 2
+THRESH_TTL = 600
+
+
+def rule(event):
+
+    # Filter events down to successful and failed login events
+    if not event.get('user_id') or event.get('event_type_id') not in [5, 6]:
+        return False
+
+    event_key = get_key(event)
+    # check risk associated with this event
+    if event.get('risk_score', 0) > 50:
+        # a failed authentication attempt with high risk score
+        if event.get('event_type_id') == 6:
+            # update a counter for this user's failed login attempts with a high risk score
+            count_failures = increment_counter(event_key)
+            if count_failures == 1:
+                set_key_expiration(event_key, time.time() + THRESH_TTL)
```

Comment:
I would say `set_key_expiration` every time without the conditional. Basically every time there is a failed suspicious login, reset the timer. Then it will only expire if they go for an hour or whatever without any suspicious logins.

What do you think?

Reply:
That's a really good point.  Without resetting with each failure, we could miss a successful login after high risk failure just because it is outside the time window when they first failed.  I'll make this change.